cluster-operator/src/main/java/io/strimzi/operator/cluster/model/JmxTrans.java

```diff
@@ -0,0 +1,393 @@
+    private static final io.strimzi.api.kafka.model.Probe DEFAULT_JMX_TRANS_PROBE = new io.strimzi.api.kafka.model.ProbeBuilder()
+            .withInitialDelaySeconds(JmxTransSpec.DEFAULT_HEALTHCHECK_DELAY)
+            .withTimeoutSeconds(JmxTransSpec.DEFAULT_HEALTHCHECK_TIMEOUT)
+            .build();
+
+    // Configuration for mounting `config.json` to be used as Config during run time of the JmxTrans
+    public static final String JMXTRANS_CONFIGMAP_KEY = "config.json";
+    public static final String JMXTRANS_VOLUME_NAME = "jmx-config";
+    public static final String CONFIG_MAP_ANNOTATION_KEY = "config-map-revision";
+    protected static final String JMX_METRICS_CONFIG_SUFFIX = "-jmxtrans-config";
+    public static final String JMX_FILE_PATH = "/var/lib/jmxtrans";
+
+    private boolean isDeployed;
+    private boolean isJmxAuthenticated;
+    private String configMapName;
+    private String clusterName;
+
+    /**
+     * Constructor
+     *
+     * @param namespace Kubernetes/OpenShift namespace where JmxTrans resources are going to be created
+     * @param kafkaCluster kafkaCluster name
+     * @param labels    labels to add to the kafkaCluster
+     */
+    protected JmxTrans(String namespace, String kafkaCluster, Labels labels) {
+        super(namespace, kafkaCluster, labels);
+        this.name = JmxTransResources.deploymentName(kafkaCluster);
+        this.clusterName = kafkaCluster;
+        this.replicas = 1;
+        this.readinessPath = "/metrics";
+        this.livenessPath = "/metrics";
+        this.readinessProbeOptions = READINESS_PROBE_OPTIONS;
+
+        this.mountPath = "/var/lib/kafka";
+
+        // Metrics must be enabled as JmxTrans is all about gathering JMX metrics from the Kafka brokers and pushing it to remote sources.
+        this.isMetricsEnabled = true;
+    }
+
+    public static JmxTrans fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup versions) {
+        JmxTrans result = null;
+        JmxTransSpec spec = kafkaAssembly.getSpec().getJmxTransSpec();
+        if (spec != null) {
+            if (kafkaAssembly.getSpec().getKafka().getJmxOptions() == null) {
+                log.info("Can't start up JmxTrans as Kafka JmxOptions is disabled");
+                return null;
+            }
+            result = new JmxTrans(kafkaAssembly.getMetadata().getNamespace(),
+                    kafkaAssembly.getMetadata().getName(),
+                    Labels.fromResource(kafkaAssembly).withKind(kafkaAssembly.getKind()));
+            result.isDeployed = true;
+
+            if (kafkaAssembly.getSpec().getKafka().getJmxOptions().getAuthentication() instanceof KafkaJmxAuthenticationPassword) {
+                result.isJmxAuthenticated = true;
+            }
+
+            result.setResources(spec.getResources());
+
+            String image = spec.getImage();
+            if (image == null) {
+                image = DEFAULT_JMXTRANS_IMAGE;
```

Comment:
You will still need to have here also the environment variable passed through the CO deployment. This is to make it easy for users to override it for example for private registries etc. So what you should do here is:
* Add the environment vaieable to the Helm Chart values & depliyment
* Run `make helm_install` to get it into the other directoties
* Check here for the environment variable and use it with following preference: 1) the `image` property from CRD, 2) The CO env. var, 3) the default image defined in this class.

Reply:
Noted! I followed TopicOperator as an example